tests/v2/waku_relay/test_wakunode_relay.nim

```diff
@@ -243,7 +247,219 @@ suite "WakuNode - Relay":
 
     await allFutures(nodes.mapIt(it.stop()))
 
-  asyncTest "Messages are relayed between two websocket nodes":
+  asyncTest "Spam protected topic accepts signed messages":
+    # Create 5 nodes
+    let nodes = toSeq(0..<5).mapIt(WakuNode.new(generateSecp256k1Key(), ValidIpAddress.init("0.0.0.0"), Port(0)))
+
+    # Protected topic and key to sign
+    let spamProtectedTopic = PubSubTopic("some-spam-protected-topic")
+    let secretKey = SkSecretKey.fromHex("5526a8990317c9b7b58d07843d270f9cd1d9aaee129294c1c478abf7261dd9e6").expect("valid key")
+    let publicKey = secretKey.toPublicKey()
+    let topicsPrivateKeys = {spamProtectedTopic: secretKey}.toTable
+
+    # Start all the nodes and mount relay with with protected topic
+    await allFutures(nodes.mapIt(it.start()))
+
+    # Mount relay with spam protected topic
+    await allFutures(nodes.mapIt(it.mountRelay(topicsPrivateKeys=topicsPrivateKeys)))
+
+    # Connect the nodes in a full mesh
+    for i in 0..<5:
+      for j in 0..<5:
+        if i == j:
+          continue
+        let conn = await nodes[i].peerManager.dialPeer(nodes[j].switch.peerInfo.toRemotePeerInfo(), WakuRelayCodec)
+        require conn.isSome
+
+    var msgReceived = 0
+    proc handler(pubsubTopic: PubsubTopic, data: WakuMessage) {.async, gcsafe.} =
+      msgReceived += 1
+
+    # Subscribe all nodes to the same topic/handler
+    for node in nodes: node.wakuRelay.subscribe(spamProtectedTopic, handler)
+    await sleepAsync(500.millis)
+
+    # Each node publishes 10 signed messages
+    for i in 0..<5:
+      for j in 0..<10:
+        let unsignedMessage = WakuMessage(
+          payload: urandom(1*(10^3)),
+          contentTopic: spamProtectedTopic,
+          version: 2,
+          timestamp: now(),
+          ephemeral: true)
+        await nodes[i].publish(spamProtectedTopic, unsignedMessage, signMessage=true)
+
+    # Wait for gossip
+    await sleepAsync(500.millis)
```

Comment:
The PR looks really great! 

While learning from it, I saw this "wait" that worries me a little. Would it be possible to make a more precise wait so that we make sure the node is in a valid "Gossip" state?

Having that kind of "waits" make me think about a possible random error, but of course I might be missing some details :)